components/net/filemanager_thread.rs

```diff
@@ -100,7 +102,16 @@ impl FileManager {
                 let filename = file_path.file_name();
 
                 match (epoch, filename) {
-                    (Ok(epoch), Some(filename)) => Some((id, Path::new(filename).to_path_buf(), epoch)),
+                    (Ok(epoch), Some(filename)) => {
+                        let filename_path = Path::new(filename);
+                        let mime = guess_mime_type_opt(filename_path);
```

Comment:
https://w3c.github.io/FileAPI/#dfn-type it said that:

> User agents can also determine the type of a Blob, especially if the byte sequence is from an on-disk file

And also, I can't image other source of type instead of determining it at the time of reading.